File: _정답코드/2주차/01_O(N^2)_정렬_2/solution.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> numbers) {
	int n = numbers.size();
    vector<int> answer = {};

    for (int iter=0; iter<n; iter++){
        int max_value = -99999;
        int max_idx = -1;

        for(int idx=0; idx < numbers.size(); idx++){
            if (numbers[idx] > max_value) {
                max_value = numbers[idx];
                max_idx = idx;
            }
        }
        
        answer.push_back(max_value);
        numbers.erase(numbers.begin() + max_idx);
    }
	return answer;
}

File: _정답코드/2주차/01_O(N^2)_정렬_2/Solution.java
import java.util.*;

public class Solution {
	public int[] solution(int[] numbers) {
        int n = numbers.length;
        ArrayList<Integer> numList = new ArrayList<>();
		ArrayList<Integer> answer = new ArrayList<>();

        for (int num : numbers){
            numList.add(num);
        }

        for (int iter = 0; iter < n; iter++){
            int maxValue = -99999;
            int maxIdx = -1;

            for (int idx = 0; idx < numList.size(); idx++){
                if (numList.get(idx) > maxValue) {
                    maxValue = numList.get(idx);
                    maxIdx = idx;
                }
            }

            answer.add(maxValue);
            numList.remove(maxIdx);
        }

		return answer.stream()
                .mapToInt(Integer::intValue)
                .toArray();
	}
}

